core/src/main/java/hudson/security/SidACL.java

```diff
@@ -147,4 +155,5 @@ protected Boolean hasPermission(Sid p, Permission permission) {
     }
 
     private static final Logger LOGGER = Logger.getLogger(SidACL.class.getName());
+    private final ConcurrentHashMap<Authentication, PrincipalSid> principalSidCache = new ConcurrentHashMap<>();
```

Comment:
> Perhaps a static cache would have more performance. I have no idea if these objects become invalid ever or if the cache would grow enormous.

@res0nance What I think Oleg meant by a static cache was that it would belong to the `SidAcl` class
```suggestion
    private static final ConcurrentHashMap<Authentication, PrincipalSid> principalSidCache = new ConcurrentHashMap<>();
```

From what I remember, the cache should still expire after a short duration because there can be external changes in the authentication service. I don't think having a `ConcurrentHashMap` here would be a good solution.

Reply:
Well then it looks like caching here is of no use since the benches would pretty much be conclusive that this is not a worthwhile idea.